koku/api/provider/models.py

```diff
@@ -105,13 +105,25 @@ class Meta:
 
     PROVIDER_AWS = 'AWS'
     PROVIDER_OCP = 'OCP'
-    PROVIDER_AZURE = 'AZURE'
+    PROVIDER_AZURE = 'Azure'
     PROVIDER_GCP = 'GCP'
     # Local Providers are for local development and testing
     PROVIDER_AWS_LOCAL = 'AWS-local'
-    PROVIDER_AZURE_LOCAL = 'AZURE-local'
+    PROVIDER_AZURE_LOCAL = 'Azure-local'
     PROVIDER_GCP_LOCAL = 'GCP-local'
 
+    PROVIDER_CASE_MAPPING = {
+        'aws': PROVIDER_AWS,
+        'ocp': PROVIDER_OCP,
+        'azure': PROVIDER_AZURE,
+        'gcp': PROVIDER_GCP,
+        'aws-local': PROVIDER_AWS_LOCAL,
+        'azure-local': PROVIDER_AZURE_LOCAL,
+        'gcp-local': PROVIDER_GCP_LOCAL,
+        'ocp-aws': 'OCP_AWS',
```

Comment:
Should we create constants for `'OCP_AWS'` and `'OCP_Azure'`, since these are used in other files?

Reply:
I agree. We just have to be careful that those constants stay out of `PROVIDER_CHOICES` and `CLOUD_PROVIDER_CHOICES` as you can't actually have an OCP_AWS provider, but we still make the distinction when it comes to our reporting APIs. I would just add a note that `OCP_AWS` and `OCP_Azure` are not actually provider types.